### PR TITLE
mingw32 compile fixes

### PR DESCRIPTION
**Pull request by emmenlau**
gcc on Windows seems to use more strict checks than Visual Studio. This PR fixes several warnings that are treated as errors with gcc on Windows (MINGW, MSYS2 and similar). The changes should be good for all Windows compilers.

### CODE REVIEW

**Comment by jtattermusch** [comment]
this is ok to test

File: include/grpc/impl/codegen/port_platform.h [hunk 1]
```diff
@@ -361,7 +361,7 @@ typedef unsigned __int64 uint64_t;
 #define GPR_MAX_ALIGNMENT 16
 
 #ifndef GRPC_MUST_USE_RESULT
-#ifdef __GNUC__
```
Comment (jtattermusch) [inline include/grpc/impl/codegen/port_platform.h]: @nicolasnoble  can you check this line?

**Comment by jtattermusch** [comment]
Thanks for the PR. All the changes LGTM, I'd like @nicolasnoble to check the change in port_platform.h

Let's see what the tests say.

**Comment by emmenlau** [comment]
Can someone help me with this PR? I'm not sure I understand the CI results, does it say the build failed for "Basic Tests Linux" and "gRPC_interop_pull_requests"? There are red crosses on those two tasks, but at the same time it says "Build finished" for both.
Furthermore, my changes should not interfere with Linux build (unless I made a stupid mistake).

**Comment by jtattermusch** [comment]
test this please

**Comment by jtattermusch** [comment]
Merged as part of #9307.

**Pull request closed without merging**